evalml/pipelines/components/transformers/imputers/simple_imputer.py

```diff
@@ -86,18 +99,24 @@ def transform(self, X, y=None):
         not_all_null_cols = [col for col in X.columns if col not in self._all_null_cols]
         original_index = X.index
 
-        # Not using select because we just need column names, not a new dataframe
-        X.ww.set_types(
-            {
-                col: "Categorical"
-                for col, ltype in X.ww.logical_types.items()
-                if isinstance(ltype, NaturalLanguage)
-            }
-        )
+        # Drop natural language columns and transform the other columns
+        X_t, natural_language_cols = self._drop_natural_language_columns(X)
+        if X_t.shape[-1] == 0:
+            return X
+        not_all_null_or_natural_language_cols = [
+            col for col in not_all_null_cols if col not in natural_language_cols
+        ]
+
+        X_t = self._component_obj.transform(X_t)
+        X_t = pd.DataFrame(X_t, columns=not_all_null_or_natural_language_cols)
+
+        # Add back in natural language columns, unchanged
+        if len(natural_language_cols) > 0:
+            X_t = pd.merge(
```

Comment:
I think `merge` and `get_subset_schema` is forcing us to re-infer the natural language columns. I think we can move `X_t.ww.init` above this line and use `ww.concat_columns`?